### PR TITLE
Fix issue with FileWrapper.getUpdated() due to missing equals() on CommittedWriterData

### PR DESCRIPTION
**Pull request by kevodwyer**

### CODE REVIEW

File: src/peergos/shared/user/CommittedWriterData.java [hunk 1]
```diff
@@ -16,4 +18,18 @@ public CommittedWriterData(MaybeMultihash hash, WriterData props) {
...
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CommittedWriterData that = (CommittedWriterData) o;
+        return Objects.equals(hash, that.hash);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(hash);
```
Comment (ianopolous) [inline src/peergos/shared/user/CommittedWriterData.java:32]: can just be hash.hashcode()

File: src/peergos/shared/user/fs/FileWrapper.java [hunk 2]
```diff
@@ -109,6 +109,13 @@ public FileWrapper withVersion(Snapshot version) {
         return new FileWrapper(capTrie, pointer, linkPointer, entryWriter, ownername, version);
     }
 
+    @JsMethod
+    public CompletableFuture<FileWrapper> getLatest(NetworkAccess network) {
+        return network.synchronizer.getValue(owner(), writer())
+                .thenCompose(finished -> getUpdated(finished, network));
```
Comment (ianopolous) [inline src/peergos/shared/user/fs/FileWrapper.java:115]: maybe current or latest is a better name than finished?

File: src/peergos/shared/user/CommittedWriterData.java (hunk 1 above)
Comment (ianopolous) [reply src/peergos/shared/user/CommittedWriterData.java:32]: I really miss non nullable types in Java

File: src/peergos/shared/user/fs/FileWrapper.java (hunk 2 above)
Comment (kevodwyer) [reply src/peergos/shared/user/fs/FileWrapper.java]: done

File: src/peergos/shared/user/CommittedWriterData.java (hunk 1 above)
Comment (kevodwyer) [reply src/peergos/shared/user/CommittedWriterData.java]: fixed

**Pull request merged**